cypress/e2e/connect_wallet.cy.ts

```diff
@@ -0,0 +1,61 @@
+describe('Connect Wallet', () => {
+	beforeEach(() => {
+		cy.interceptCurrentUserRequest();
+		cy.visitRootDomain();
+	});
+
+	afterEach(() => {
+		// reset connection for following test
+		cy.disconnectMetamaskWalletFromDapp().then((disconnected) => {
+			expect(disconnected).to.be.true;
+		});
+	});
+
+	after(() => {
+		cy.resetMetamaskAccount();
+	});
+
+	it('successfully opens the connect wallet modal', () => {
+		cy.get('.authentication__connect-wrapper').click();
+		cy.get('div').should('have.class', 'wallet-select__header');
+		cy.contains('connect to a wallet', { matchCase: false }).should(
+			'be.visible',
+		);
+		cy.get('.dialog__content')
+			.should('contain', 'Metamask')
+			.and('contain', 'Wallet Connect')
+			.and('contain', 'Coinbase Wallet')
+			.and('contain', 'Fortmatic')
+			.and('contain', 'Portis');
```

Comment:
Is it possible to write these in a way which is more similar to how the user would navigate the website?

i.e. `find a button with the text "Connect To A Wallet", click it, check if there's a modal with the text "Metamask", "Wallet Connect", etc.`

This way we're not tied to implementation details, making tests less brittle.

Similar comments for other tests - would be good if we could use `getByText` as much as possible although I know that can be difficult!

Reply:
Yes for sure 🙂 working to improve these now in the follow up PR